apps/emqx_modules/src/emqx_modules_app.erl

```diff
@@ -33,7 +33,21 @@ stop(_State) ->
     ok.
 
 maybe_enable_modules() ->
-    emqx_conf:get([delayed, enable], true) andalso emqx_delayed:enable(),
+    DelayedEnabled = emqx_conf:get([delayed, enable], true),
+    RewriteEnabled = length(emqx_conf:get([rewrite], [])) > 0,
+    RetainerEnabled = emqx_conf:get([retainer, enable], false),
+    AutoSubscribeEnabled = length(emqx_conf:get([auto_subscribe, topics], [])) > 0,
+    application:set_env(
```

Comment:
Set env for this app and expose an api for telemetry to access

Reply:
nit: call the set api

Reply:
D'oh, changed other places to use it and forgot this one... :see_no_evil:

Reply:
I'll fix it in a following PR.